src/rpc/protocol.h

```diff
@@ -29,7 +29,7 @@ enum HTTPStatusCode
 };
 
 //! Bitcoin RPC error codes
-enum RPCErrorCode
+enum class RPCErrorCode
```

Comment:
I still think that ```RPCErrorCode``` and ```HTTPStatusCode``` are intrinsically different than the other enumerators, since they are sent over HTTP. They are also the only two that have explicit values assigned. They should be treated as integers always and they should require no conversion.

I would have suggested to hide conversion in encode/decode functions. Then the conversion would only have to be done once after receiving an http status. But then I see ```response.status >= 400``` in ```bitcoin-cli.cpp```. This suggests that HTTP status codes are not treated like enumerators and should not be compared to enumerators either.

```RPCErrorCode``` and ```HTTPStatusCode``` should be treated more like macros instead.
Strong typing them would case unnecessary complexity due to the casting. But that would only be a problem if they begin to be used much more often, which is speculation on my part. So these changes are fine for now.